### ISSUE TITLE
Write-time target and sibling checks resolve "does a skill exist?" against the session's visible scope only, so project-scoped skills in other repositories are filed as new-skill proposals

### ISSUE THREAD

**Issue by MiltonSilvaJr** [rebelytics/one-skill-to-rule-them-all#213]
**Target:** `SKILL.md` — "Validate the target at write time" / "Check the target's siblings at write time"; `references/observation-log.md` — "Skill families and the sibling check" (the no-registry fallback); `references/weekly-review.md` — Step 3, "Consolidate new-skill candidates".
**Verified at:** upstream `main` (`80ab2f3`, v3.3.1). The write-time checks still resolve against "the installed skill names", and the no-registry fallback still says to "scan the installed skill names for a shared prefix, suffix or subject". Nothing asks which scopes that set covers.
**Adjacent, not duplicate:**
- #143 (closed) added to Session Start step 6: "Say what you resolved against (this checkout, this install): an unresolved target is a fact about where you looked, not about the world." That is the right principle, but it landed only in step 6's target resolution, and its failure case is a *stale checkout*. This report is about a different cause, *scope*, and about the write-time and review-time checks, which carry no such sentence.
- #144 (closed) gave the review a way to correct a wrongly recorded target after the fact. This report is the prevention side: the same wrong `proposes_skill:` is cheap to avoid at write time, and the review should look before it presents a new-skill cluster rather than only after a correction becomes necessary.
- #59 (closed) covers checking an *upstream project* for an official skill before authoring one. This is about skills that already exist on the same machine, in scopes the current session cannot see.

## What happened

One session, working in project A, logged four observations about a workflow (keeping a fork in sync with its upstream). Each one recorded `proposes_skill: [<new-name>]` and `siblings_checked: none`, on the honest judgement that no installed skill covered the topic.

A skill covering exactly that workflow already existed, as a project-scoped skill in two *other* repositories on the same machine: in each one as a generator source (a tool-owned directory that renders into `.claude/skills/`) and as the rendered copy. It was also the target of a two-member family. Two of the four observations were already applied in its text. The weekly review avoided re-applying them on top only because it happened to run a filesystem search across the projects root and then a presence check on the body it found.

## Why the current text did not prevent it

The set of skills a session can see is the user scope plus the project the session runs in. The write-time rules treat that set as "the installed skills":

> Validate the target at write time. A name in `skill:` must be a skill that exists now; if it doesn't, the observation proposes a skill instead.

> No registry yet, or the target is not in it? The check is still required: scan the installed skill names for a shared prefix, suffix or subject …

Both run correctly and give a confident wrong answer, because the question ("does a skill for this exist?") is asked of a set that leaves out every project-scoped skill outside the current project and every generator source those skills are rendered from. Nothing records *where* the check looked, so `siblings_checked: none` reads the same whether the check covered one scope or all of them. The review's "Consolidate new-skill candidates by the problem they solve" merges proposals *with each other*, but never checks them against existing skills in other scopes.

The failure only goes one way: a narrow scope never invents a skill, it only misses real ones. So it always produces the costly outcome: new-skill proposals for work that already exists, and observations that never reach the skill they belong to.

## Proposed fix

Three additions. All of them add text and remove nothing. I have run them locally since this week.

**1. `SKILL.md`, after "Validate the target at write time":**

> **An absence claim inherits the scope of the search that produced it.** The skills visible in a session are the user scope plus the project the session runs in, not every skill that exists. Skills scoped to other projects, and the generator sources that produce project copies (a tool-owned directory rendered into `.claude/skills/`, for example), are invisible from here, so "no installed skill covers this" is a blind spot of the observer until those scopes have been searched. Before writing a `proposes_skill:`, or `none` in `siblings_checked:`, run the project-scope sweep (`references/observation-log.md`, sibling check) and name in `siblings_checked:` which scopes were searched. "Not found" is only a finding together with where you looked.

**2. `references/observation-log.md`, in "Skill families and the sibling check", after the two cheap tests:**

> **Sweep the project scopes before asserting absence.** The installed skill names a session can see cover the user scope and the current project only; the same skill often lives in other projects' `.claude/skills/` and in the generator sources those copies are rendered from. Before recording a target as missing, a family as empty, or a `proposes_skill:` as new, enumerate the project scopes by glob and guard the enumeration, because an empty result tells you about the instrument before it tells you about the library. Avoid a recursive `find` here: over a projects root full of worktrees and dependency trees it can run for minutes.
>
> ```bash
> root="[PROJECTS ROOT]"   # absolute path of the directory holding the user's repositories
> list=$(for d in "$root"/*/.claude/skills/*/ "$root"/*/[TOOL-OWNED SKILLS DIR]/*/ "$root"/*/*/.claude/skills/*/ "$root"/*/*/[TOOL-OWNED SKILLS DIR]/*/; do [ -f "$d/SKILL.md" ] && echo "${d%/}"; done)
> n=$(printf "%s\n" "$list" | grep -c .)
> [ "$n" -eq 0 ] && { echo "SWEEP BROKEN — no project skill found under $root"; exit 1; }
> printf "%s\n" "$list" | grep "/<skill-name>$"                                            # by name
> printf "%s\n" "$list" | while IFS= read -r d; do grep -Il -i -- "<term from the body>" "$d/SKILL.md" || :; done   # by problem; a non-match is not an error
> ```
>
> Search by the problem as well as by the name: a proposal is named after the task that surfaced it, and an existing skill that already solves it rarely has the same name. Add further scope directories to the globs where the environment keeps skills elsewhere. Record the scopes searched in `siblings_checked:`.

**3. `references/weekly-review.md`, Step 3, as a new item next to "Consolidate new-skill candidates":**

> **Existence check before any new-skill cluster is presented.** A `proposes_skill:` or a `siblings_checked: none` written from one session's view is an absence claim limited to that session. Re-run the project-scope sweep (`observation-log.md`, sibling check) using the problem terms in the bodies, not only the proposed name, and run the presence check of Step 5 against any skill it finds. An observation proposing a skill that already exists elsewhere is an improvement to that skill, and may already be applied there.

What I measured with (2), in a clean shell against a real projects root:
- It listed several hundred project skills.
- It found all four known copies of the skill described above (two sources, two rendered copies).
- A root that does not exist triggered the `SWEEP BROKEN` guard.

A first version used `find <root> -maxdepth 5 -type d -path '*skills/<name>'`. It ran for more than 120 s on the same tree, which is why the snippet uses globs.

## Principle

An absence claim inherits the scope of the search that produced it. When artefacts of the same kind live in several scopes (user, project, generator source versus rendered copy), a search that covers only the current session's scope turns the observer's blind spot into a recorded fact. #143 states this for one step. It holds for every check in the skill that answers "does it exist?".
